### DISCUSSION TITLE
[FEATURE REQ] Filters for custom torznab attributes (tag, etc.)

### DISCUSSION THREAD

**Discussion by gnir-gh** [autobrr/autobrr#1683] [Ideas]
**Is your feature request related to a problem? Please describe.**
Torznab feeds sometimes contain [extended attributes](https://torznab.github.io/spec-1.3-draft/torznab/Specification-v1.3.html#extended-attributes) that can be very useful for filtering. For example, "tag" named attribute is sometimes used to indicate 2x upload torrents, for example. Something like

`<torznab:attr name="tag"  value="doubleupload" />`

Other potentially useful attributes include the number of seeders ("seeders" attribute) and leechers ("leecher" attribute) and finished downloads ("grab" attribute) at grab time, e.g., for checking whether you are early in a swarm. Since the availability of these attributes varies from tracker to tracker, it would be nice to have a custom solution that users can tailor for specific trackers in each filter.

**Describe the solution you'd like**
A new section in a filter that allows users to define the attribute(s) and value(s), preferably with an option for regex matching.

**Describe alternatives you've considered**
A more user-friendly alternative is to parse only specific attributes, and have a separate filter option for each. However, I prefer the flexibility that my proposal offers. The middle-ground approach is to give a widely available attribute (e.g., seeders) its own section while allowing custom attributes.

**Comment by chiva** [comment]
Very interested on this too.

I think having some very advanced option to use xpath or jq can greatly extend the flexibility of the software, so that we can use the tool on more complex scenarios without having to wait for the exact feature to be developed

**Comment by SysTechIL** [comment]
This is exactly what I'm looking for!
I want to adjust to:
<torznab:attr name="uploadvolumefactor" value="2" />
This means that "uploadvolumefactor" should be equal to "2" but I haven't found a way to set this in Autobrr.
I would love to see this option added.

**Comment by zze0s** [comment]
Related discussion with some more info https://github.com/autobrr/autobrr/discussions/2157
